components/autocomplete/Autocomplete.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import Awesomplete from 'awesomplete';
+import SelectedItems from './SelectedItems';
+import './Autocomplete.scss';
+
+const createInputEventListener = (inputRef) => (event, callback, deps) => {
+    useEffect(() => {
+        inputRef.current.addEventListener(event, callback);
+
+        return () => {
+            inputRef.current.removeEventListener(event, callback);
+        };
+    }, deps);
+};
+
+// TODO: move state to parent (hook?)
+const Autocomplete = ({
+    value,
+    multiple,
+    onSelect = () => {},
+    onOpen = () => {},
+    onClose = () => {},
+    onHighlight = () => {},
+    ...rest
+}) => {
+    const inputRef = useRef(null);
+    const containerRef = useRef(null);
+    const awesompleteEventListener = createInputEventListener(inputRef);
+    const [inputValue, setInputValue] = useState(multiple ? '' : value);
+    const [selected, setSelected] = useState(value ? [].concat(value) : []);
+
+    useEffect(() => {
+        new Awesomplete(inputRef.current, {
```

Comment:
Here you need to save `Awesomplete` instance when we unmount the component and call `destroy()`.

``` js
const awesomplete = new Awesomplete();

return () => {
   awesomplete.destroy();
};
```